### PR TITLE
createkubertetesbinariesiso: Saving images in network and dashboard yaml

### PR DESCRIPTION
**Pull request by davidjumani**
## Description
Downloads images int the network and dashboard yaml while creating the kubernetes binaries iso
This step was skipped, so bringing up a cluster on a node that does not have access to the internet failed since the weavenet and dashboard images were not found locally

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
```
./create-kubernetes-binaries-iso.sh ./ 1.11.4 0.7.1 1.11.1 https://github.com/weaveworks/weave/releases/download/latest_release/weave-daemonset-k8s-1.11.yaml https://raw.githubusercontent.com/kubernetes/dashboard/v1.10.0/src/deploy/recommended/kubernetes-dashboard.yaml
..........
Downloading docker image k8s.gcr.io/kube-apiserver-amd64:v1.11.4 ---
Downloading docker image k8s.gcr.io/kube-controller-manager-amd64:v1.11.4 ---
Downloading docker image k8s.gcr.io/kube-scheduler-amd64:v1.11.4 ---
Downloading docker image k8s.gcr.io/kube-proxy-amd64:v1.11.4 ---
Downloading docker image k8s.gcr.io/pause:3.1 ---
Downloading docker image k8s.gcr.io/etcd-amd64:3.2.18 ---
Downloading docker image k8s.gcr.io/coredns:1.1.3 ---
Downloading docker image weaveworks/weave-kube:2.7.0 ---                       << Added image from network yaml
Downloading docker image weaveworks/weave-npc:2.7.0 ---                        << Added image from network yaml
Downloading docker image k8s.gcr.io/kubernetes-dashboard-amd64:v1.10.0 ---     << Added image from dashboard yaml
```

### CODE REVIEW

**Comment by shwstppr** [comment]
Nice! Thanks @davidjumani. Will test this out

**Comment by shwstppr** [comment]
@davidjumani should this go in 4.14? And do we need to make any change in docs (added a new param for script)?

File: scripts/util/create-kubernetes-binaries-iso.sh [hunk 1]
```diff
@@ -87,6 +91,14 @@ if [ $? -ne 0 ]; then
 fi
 mkdir -p "${working_dir}/docker"
 output=`${k8s_dir}/kubeadm config images list --kubernetes-version=${RELEASE}`
+
+# Don't forget about the yaml images !
+for i in ${network_conf_file} ${dashboard_conf_file}
+do
+  images=`grep "image:" $i | cut -d ':' -f2- | tr -d ' ' | tr -d "'"`
+  output=`printf "%s\n" ${output} ${images}`
```
Comment (DaanHoogland) [inline scripts/util/create-kubernetes-binaries-iso.sh:99]: `printf` has only one specifier in the format and two arguments. What is intended here?

**Review by DaanHoogland — COMMENTED** [review]
one question

File: scripts/util/create-kubernetes-binaries-iso.sh (hunk 1 above)
Comment (davidjumani) [reply scripts/util/create-kubernetes-binaries-iso.sh:99]: It's a simpler way of doing `printf "%s\n%s\n" ${output} ${images}` :D

**Comment by davidjumani** [comment]
> @davidjumani should this go in 4.14? And do we need to make any change in docs (added a new param for script)?

@shwstppr Don't think it's needed as it's an optional parameter and used in automated build jobs

File: scripts/util/create-kubernetes-binaries-iso.sh (hunk 1 above)
Comment (DaanHoogland) [reply scripts/util/create-kubernetes-binaries-iso.sh:99]: ok, never encountered that before, tnx

File: scripts/util/create-kubernetes-binaries-iso.sh [hunk 2]
```diff
@@ -16,9 +16,11 @@
 # specific language governing permissions and limitations
 # under the License.
 
+set -e
```
Comment (DaanHoogland) [inline scripts/util/create-kubernetes-binaries-iso.sh:19]: :tr: red flag ;) but fine by me. we allow this script to exit and leave garbage behind. in view of the function of this script fine. (a comment or so would be appropriate, i think)

**Review by DaanHoogland — APPROVED** [review]
looks good

**Review by shwstppr — APPROVED** [review]
LGTM

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Pull request merged**